marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/writer/HubDataWriter.java

```diff
@@ -114,4 +121,41 @@ private String convertRowToJSONString(InternalRow record) {
         return jsonObjectWriter.toString();
     }
 
+    protected JsonNode determineIngestionEndpointParams(Map<String, String> params) {
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        ObjectNode endpointParams;
+        if (params.containsKey("ingestendpointparams")) {
+            try {
+                endpointParams = (ObjectNode)objectMapper.readTree(params.get("ingestendpointparams"));
+            } catch (IOException e) {
+                throw new RuntimeException("Unable to parse ingestendpointparams, cause: " + e.getMessage(), e);
+            }
+        } else {
+            endpointParams = objectMapper.createObjectNode();
+        }
+
+        boolean doesNotHaveApiPath = (!endpointParams.hasNonNull("apiPath") || StringUtils.isEmpty(endpointParams.get("apiPath").asText()));
+        boolean hasWorkUnitOrEndpointState = endpointParams.has("workUnit") || endpointParams.has("endpointState");
+        if (doesNotHaveApiPath && hasWorkUnitOrEndpointState) {
+            throw new RuntimeException("Cannot set workUnit or endpointState in ingestionendpointparams unless apiPath is defined as well.");
+        }
+
+        if (!endpointParams.hasNonNull("apiPath") || StringUtils.isEmpty(endpointParams.get("apiPath").asText())) {
+            endpointParams.put("apiPath", "/data-hub/5/data-services/ingestion/bulkIngester.api");
+        }
+
+        // TODO : remove the below else block after java-client-api 5.3 release
```

Comment:
else block missing

Reply:
There's no else needed - this is just saying "If Ernie didn't provide a non-null endpointState, then toss in an empty object node".